src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -625,50 +673,112 @@ impl AccessChecker for PrivilegeAccess {
 
             // Database.
             Plan::ShowCreateDatabase(plan) => {
-                self.validate_db_access(&plan.catalog, &plan.database, UserPrivilegeType::Select, false).await?
+                self.validate_db_access(
+                    &plan.catalog,
+                    &plan.database,
+                    UserPrivilegeType::Select,
+                    false,
+                )
+                .await?
             }
             Plan::CreateDatabase(_) => {
                 self.validate_access(&GrantObject::Global, UserPrivilegeType::CreateDatabase)
                     .await?;
             }
+            // Task
+            // Is it compatible with the permission check of the old version?
             Plan::CreateTask(_) => {
-                self.validate_access(&GrantObject::Global, UserPrivilegeType::CreateTask)
-                    .await?;
+                if self
+                    .validate_access(&GrantObject::Global, UserPrivilegeType::CreateTask)
+                    .await
+                    .is_err()
+                {
+                    self.validate_access(&GrantObject::Global, UserPrivilegeType::Super)
+                        .await?;
+                }
             }
             Plan::AlterTask(plan) => {
-                self.validate_task_access(&plan.task_name, UserPrivilegeType::Alter).await?;
+                if self
+                    .validate_task_access(&plan.task_name, UserPrivilegeType::Alter)
+                    .await
+                    .is_err()
+                {
+                    self.validate_access(&GrantObject::Global, UserPrivilegeType::Super)
+                        .await?;
+                }
             }
             Plan::DropTask(plan) => {
-                self.validate_task_access(&plan.task_name, UserPrivilegeType::Drop).await?;
+                if self
+                    .validate_task_access(&plan.task_name, UserPrivilegeType::Drop)
+                    .await
+                    .is_err()
+                {
+                    self.validate_access(&GrantObject::Global, UserPrivilegeType::Super)
+                        .await?;
+                }
+            }
+            Plan::DescribeTask(plan) => {
+                let session = self.ctx.get_current_session();
+                if self
+                    .has_ownership(&session, &GrantObject::Task(plan.task_name.to_owned()))
+                    .await
+                    .is_err()
+                {
+                    self.validate_access(&GrantObject::Global, UserPrivilegeType::Super)
+                        .await?;
+                }
+            }
+            Plan::ExecuteTask(plan) => {
```

Comment:
Maybe execute/describe task only need to check the user's ownership ?

Reply:
yeah, i think so

Reply:
Ok